sympy/core/relational.py

```diff
@@ -109,13 +119,16 @@ def canonical(self):
         """
         r = self
         if r.func in (Ge, Gt):
-            r = r.reversed
+            if _coeff_isneg(r.rhs):
```

Comment:
I thought this modification was necessary to avoid an infinite recursion that I was getting where  I was getting alternation between `x**2 > -y` and `y > -x**2`. But I see in master that the canonical form is the latter (as we would expect).

As to your sentiments regarding what looks right, I tried to respect those with the former introduction of the canonical method:

```
>>> from sympy.core.relational import _canonical as f  # new function to make expr canonical
>>> f(x>a)
a < x
>>> f(x>a+b)
x > a + b
>>> f(x>a+1)
x > a + 1
>>> f(Or(x>b,x<a))
(b < x) | (x < a)
>>> f(Or(x<a,x>b))
(b < x) | (x < a)
>>> f(And(a<x,x<b))
(a < x) & (x < b)
```

I'll remove this modification and try better locate the source of the recursion.